cluster-autoscaler/cloudprovider/builder/cloud_provider_builder.go

```diff
@@ -35,17 +35,19 @@ import (
 // CloudProviderBuilder builds a cloud provider from all the necessary parameters including the name of a cloud provider e.g. aws, gce
 // and the path to a config file
 type CloudProviderBuilder struct {
-	cloudProviderFlag string
-	cloudConfig       string
-	clusterName       string
+	cloudProviderFlag       string
+	cloudConfig             string
+	clusterName             string
+	autoprovisioningEnabled bool
 }
 
 // NewCloudProviderBuilder builds a new builder from static settings
-func NewCloudProviderBuilder(cloudProviderFlag string, cloudConfig string, clusterName string) CloudProviderBuilder {
+func NewCloudProviderBuilder(cloudProviderFlag string, cloudConfig string, clusterName string, autoprovisioningEnabled bool) CloudProviderBuilder {
```

Comment:
Maybe we should just pass AutoscalingOptions here instead of a semi-random subset of flags? It feels cleaner and more generic.

However, it would require moving AutoscalingOptions to a separate file to avoid problems with cyclic imports (and this PR is complex as it is). @mwielgus WDYT? @krzysztof-jastrzebski maybe you want to tackle that as part of your PR if @mwielgus agrees?